### PR TITLE
Allow 0 for Network Watcher Flow Logs retention days

### LINKED ISSUE #1055: CKV_AZURE_12: Retention policy of 0 is not supported
**Describe the bug**
When setting the `retention_period.days` value on `azurerm_network_watcher_flow_log` to `0`, `CKV_AZURE_12` still fails. `0` is the value to signify indefinite/forever retention.

**To Reproduce**
1. Create an `azurerm_network_watcher_flow_log` resource
2. Set
```
  retention_policy {
    enabled = true
    days    = 0
  }
```
3. Run `checkov`
4. Receive failure for `CKV_AZURE_12`

**Expected behavior**
`0` is an accepted value as documented at https://docs.microsoft.com/en-us/azure/network-watcher/network-watcher-nsg-flow-logging-overview#how-logging-works

**Desktop (please complete the following information):**
 - OS: Linux
 - Checkov Version: 2.0.26

**Additional context**
At https://github.com/bridgecrewio/checkov/blob/master/checkov/terraform/checks/resource/azure/NetworkWatcherFlowLogPeriod.py#L17 likely need to adjust the innermost `if` statement to something like:

```python
if retention_in_days is not None and (retention_days == 0 or retention_days >= 90):
```

Happy to open a pull request if this is an acceptable solution.

### PR DESCRIPTION
**Pull request by kylelaker**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Closes #1055

### CODE REVIEW

**Review by arielkru — COMMENTED** [review]
Good catch, thanks :)
I think we should also handle cases where `days` is set to a string (not a valid integer, as you proposed), take a look at my comments :)

File: checkov/terraform/checks/resource/azure/NetworkWatcherFlowLogPeriod.py [hunk 1]
```diff
@@ -16,7 +16,9 @@ def scan_resource_conf(self, conf):
             retention_block = conf['retention_policy'][0]
             if retention_block['enabled'][0]:
                 retention_in_days = force_int(retention_block['days'][0])
-                if retention_in_days and retention_in_days >= 90:
+                # Comparing to 0 is necessary since `force_int` may return None (falsey) if
+                # "days" is not a valid integer.
+                if retention_in_days == 0 or retention_in_days >= 90:
```
[the remark this answers is not in the document]
Comment (kylelaker) [reply checkov/terraform/checks/resource/azure/NetworkWatcherFlowLogPeriod.py:21]: I am not sure that handling `None` in this way would be correct. Omitting `days` would cause the above line that does `['days']` to raise a `KeyError` (perhaps that should be `.get('days', [None])`?). If `days` is present but cannot be converted to an `int`, then `force_int` will have returned `None` and therefore, this expression will evaluate to `False`, the condition will fail, and we'll fall to the `return CheckResult.FAILED` below.

Accepting `None` as a `PASSED` state would be explicitly _incorrect_ because it means that `days` is a totally invalid value and is especially not `0` or a number >= 90.

The only correct way to handle `None` would be a second `if retention_in_days is None: return CheckResult.FAILED` but that's redundant with the final statement in the function.

As it is, omitting `days` does cause an odd `KeyError` failure in the tests; however, it does return what I would expect to be correct error in the CLI execution as a whole, reporting that it failed to run the check.

Basically, the check _should not_ pass without `days` present, because per https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/network_watcher_flow_log#days, `days` is a required value. There is not a valid "default" value per the documentation. If there were, then perhaps accepting `None` would be correct but to not provide `days` results in an invalid Terraform resource.

Is there a better way to reword the comment that would make this more clear? Should the comment highlight that `None` is not a passing value?

[the remark this answers is not in the document]
Comment (kylelaker) [reply checkov/terraform/checks/resource/azure/NetworkWatcherFlowLogPeriod.py]: Updated the code to _explicitly_ reject `None`. Hopefully this makes the intent clearer. It leaves the `KeyError` as a possibility but that would have happened regardless. Could be made to be `.get('days', [None])` if desired but I don't know if that would be correct since the resource would be invalid.

Otherwise, hopefully explicitly rejecting `None` makes this a little bit clearer with the intent.

File: tests/terraform/checks/resource/azure/test_NetworkWatcherFlowLogPeriod.py [hunk 2]
```diff
@@ -48,6 +48,26 @@ def test_success(self):
...
+                enabled = true
+                days    = 0
+              }
+              }
+
+                """)
+        resource_conf = hcl_res['resource'][0]['azurerm_network_watcher_flow_log']['test']
+        scan_result = check.scan_resource_conf(conf=resource_conf)
+        self.assertEqual(CheckResult.PASSED, scan_result)
+
```
[the remark this answers is not in the document]
Comment (kylelaker) [reply tests/terraform/checks/resource/azure/test_NetworkWatcherFlowLogPeriod.py:90]: Added a test for both the possibility of a valid and for an invalid integer as a `string`.

**Review by nimrodkor — APPROVED** [review]
